### PR TITLE
[Duplicate] The Best Frameworks for Developing Decentralized Applications

### LINKED ISSUE #5239: The Best Frameworks for Developing Decentralized Applications
*NOTE*: All **1st-time contributors** should know Topic approval and PR submission does NOT guarantee your Topic/Article will be published. Our team of Peer Reviewers and Content Moderators will review all PRs that come in to make sure they adhere to the standard of quality we expect from the EngEd community. 

We expect all community memebers to go through provided examples, resources, and previously published material before submitting content. As a rule of thumb, please only submit articles (pull requests) that are complete, formatted correctly, and include a fully polished article (ready to be published and error free). 

All revisions and edits should be *completed on your own forked repo (as to not take up room in the queue)*. **Any PR that is submitted incomplete (meaning not ready to be published as is) will be closed.** A PR being closed does NOT mean the article can not be published (or fixed) it simply means that the edits, revisions, and fixes will happen outside of the queue.

### Topic Suggestion
The Best Frameworks for Developing Decentralized Applications

#### Pre-submission advice
By following all our pre-submission advice and reviewing our Resources folder, you will maximise your chances of your topic being approved.

We ask that you please be patient as our team works through approving and publishing all articles/tutorials in a timely manner.

Allow 1-3 days for a topic to be reviewed and/or approved - allow 3-7 days for an articles to be reviewed and/or published (subject to vary depending on _volume and/or backlog of articles_).

Be sure to visit our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for tools, resources, and example articles that will help you propose and write a successful article.

**Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**

**We tend to stray away or tend not to publish reviews/comparisons of commercial product offerings.**

### Writing sample(s): 
Include any links or writing samples - to help our team better gauge your writing quality.

#### Proposed title of article
The Best Frameworks for Developing Decentralized Applications

#### Proposed article introduction
With the rise of blockchain technology, which brings decentralization, more decentralized applications (DApps) are being developed. Frameworks specifically created to simplify DApp development are becoming more and more common. DApp frameworks, like other frameworks, are pieces of code written to the software, making it easy to build blockchain applications. Frameworks can help you develop better and faster apps.

This article will look at the popular frameworks used by blockchain developers to build DApps. Also, it will look at what are DApps, why DApps are the key to future technology, and the different types of DApps.

#### Key takeaways 
1. What are Decentralized Applications (DApps)
2. Why DApps are key to the future technology
3. Types of DApps
- DEFI
- DAOs
- NFTs
- Tracking Apps
4. DApp frameworks

- Truffle
- Brownie
- Hardhat
- Web3j
- Open Zeppelin
- The Graph
- Embark

#### Article quality
This article will help you understand DApps and their benefits in detail. It will cover in-depth the DApps frameworks, what the framework does, why you'd want to use it, why it's one of the best, and anything else noteworthy.

#### References
Please list links to any published content/research that you intend to use to support/guide this article.

### Conclusion
Finally, remove the Pre-Submission advice section and all our blockquoted notes as you fill in the form before you submit.
We look forwarding to reviewing your topic suggestion.

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by julie-techie**
## Our **Peer Reviewer and Content Moderator teams do NOT provide any revisions services.** All revisions and edits should be completed on your own forked repo (as to not take up room in the queue) in order for our team to review them all in a timely manner. 

**ANY ARTICLE SUBMITTED WITH GLARING ERRORS WILL BE IMMEDIATELY CLOSED.**

>As a rule of thumb - please be sure to only submit articles (pull requests) that are fully polished and ready to be published. Be sure to go through our resources documents for extra and 3rd party (vetted) resources to help improve overall technical writing.

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is **600x400 pixels**, under **300KB in size**, and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes #5239

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

**Comment by julie-techie** [comment]
@mbaabuones have made the necessary changes and opened an new PRwhich is a duplicate of this #5603

**Comment by Qodestackr** [comment]
Hey choose between dApps or DApps, just for consistency.👍

**Comment by srishilesh** [comment]
@mbaabuones 

The author has been waiting for quite a while for the review of this duplicate PR. Assigning it to you.

File: content/articles/best-frameworks-for-developing-dapps/index.md [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+### Prerequisites
+To follow this guide, you'll need to be familiar with blockchain, software development, and [Web 3.O](engineering-education/overview-of-web-3-0/).
+
+### Decentralized Applications (DApps)
+Applications such as Google, Twitter, Facebook, and PayPal run on systems created and operated by the company. These applications run on a centralized server. As a result, the organizations have complete control over the apps and their functionality. A Decentralized application (DApp) is very similar in how it works, except that instead of reporting back to Google or PayPal servers, this app reports to the blockchain. You interact with the blockchain through the application.
+
+DApps are systems that run on peer-to-peer and blockchain networks instead of central servers. Because DApps are powered by blockchain, they are different from other applications and perhaps more superior.
+
+These features make dApps essential in the current and future world.
++ All dApps operations are stored on a blockchain to stimulate productivity, trust, and transparency.
```
Comment (mbaabuones) [inline content/articles/best-frameworks-for-developing-dapps/index.md:21]: Here, you can put the key words in bold. Provide a brief explanation after the key words

File: content/articles/best-frameworks-for-developing-dapps/index.md [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+
+### Why DApps are key to the future technology
+DApp is a fascinating technology that has yet to be fully exploited. DApps are set to help the world due to their tone of benefits and features. Dapps have a lot of potentials to transform the way we work, communicate, do transactions, and much more.
+
+Dapps will continue to grow tremendously in the future. They will become increasingly involved in our everyday life, lowering costs and removing intermediaries from many of our activities.
+
+We expect that Dapps will keep increasing as the adoption of blockchain technology will bring a lot of benefits to various sectors. These sectors include finance, healthcare blockchain, Education, Gaming, and more.
+
+DApps exciting benefits include:
++ Censorship resistant - Users can control their activities because dapps are more sovereign.
```
Comment (mbaabuones) [inline content/articles/best-frameworks-for-developing-dapps/index.md:35]: Put the key words in bold

File: content/articles/best-frameworks-for-developing-dapps/index.md [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+Go through the [Truffle documentation](https://trufflesuite.com/docs/truffle/overview).
+
+#### Brownie
+Brownie is a Python-based framework for building and testing smart contracts targeting the [Etherium Virtual Machine (EVM)](https://ethereum.org/en/developers/docs/evm/) applications. It's python-based, meaning it uses python libraries such as web3.py and pytest. Also, it uses python to write scripts. 
+Brownie is a powerful and easy-to-use framework that handles deployment, debugging and testing of dApps.
+
+Here is some feature of Brownie:
++ There is [Solidity](https://github.com/ethereum/solidity) and [Vyper](https://github.com/vyperlang/vyper) support.
++ For easy project engagement, there's a built-in console.
++ Contract testing using [pytest](https://github.com/pytest-dev/pytest).
```
Comment (mbaabuones) [inline content/articles/best-frameworks-for-developing-dapps/index.md:90]: Explain this statement and the other two statements.

**Review by mbaabuones — CHANGES_REQUESTED** [review]
@julie-techie Use either DApp or dApp consistently. Please work on the suggested minor changes and ping me when you are done.

**Comment by julie-techie** [comment]
@mbaabuones, I made the changes you suggested

**Review by mbaabuones — APPROVED** [review]

**Comment by Kayere** [comment]
Great article @julie-techie
Thank you for the review @mbaabuones

**Pull request merged**